pkg/authentication/interfaces.go

```diff
@@ -0,0 +1,40 @@
+// Copyright 2024 Canonical Ltd.
+// SPDX-License-Identifier: AGPL-3.0
+
+package authentication
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/coreos/go-oidc/v3/oidc"
+	"golang.org/x/oauth2"
+)
+
+type providerVerifier interface {
+	Verify(ctx context.Context, rawIDToken string) (*oidc.IDToken, error)
+}
+
+type TokenVerifier interface {
+	VerifyAccessToken(ctx context.Context, rawAccessToken string) (*Principal, error)
```

Comment:
I'm open to drop it since it's clear anyway
I would usually prefer to have names when multiple parameters are present like multiple strings, to allow distinguishing.
Changing it in next revision